src/main/java/org/passay/RuleResult.java

```diff
@@ -23,8 +24,13 @@
   protected RuleResultMetadata metadata = new RuleResultMetadata();
 
 
-  /** Default constructor. */
-  public RuleResult() {}
+  /**
+   * Creates a new rule result with its validity set to true.
+   */
+  public RuleResult()
+  {
+    this(true);
```

Comment:
I originally preferred `new RuleResult(true)` as it served as a reminder that the result you're dealing with is a success unless you toggle it to a failure. Having worked with the API for a while I can't say I feel strongly about it anymore. Unfortunately we included a default constructor which left `valid` as `false`, so this would be a breaking change for any users of the API.

This change would necessitate a version bump to 1.5.0 or more likely to 2.0.0 and probably nice warning message in the docs.

Reply:
So do you want me to move this commit to a separate PR? Or you'll merge it to master which will become a 2.0.0 in any case? Whatever you prefer.

Reply:
Or revert to using the boolean true everywhere? Or add a comment to the constructor javadoc about the change?

Reply:
Unless you want a 1.4.2 built for Java 8, I'll push 2.0.0 into master.

Reply:
I understand you need a version bump due to the API changes, but don't quite get how this relates to the JDK version - do you intend to change the source/target level as well? is there anything in the current code that doesn't work on newer JDKs? Or something you want to use in newer JDK that isn't available in 8?

Reply:
We're in the process of moving projects to Java 11, so no reason other than it's on my roadmap to do. Source and target will be changed to 11. Current code appears to work just fine on Java 11.

Reply:
Doesn't that mean it will no longer run on the Java 8 JVM? Since the majority of Java systems still use 8, won't it be better for a library that doesn't need newer features to stick with source/target 8 (which will run on both 8 and newer versions)? Might be worth considering, especially if there's no real need to upgrade target to 11.

Reply:
With Java8 EOL for commercial customers, it will be interesting to see how quickly people move to 11. (And Amazon Corretto may apply some headwind as well...) For now I'll build with OpenJDK 11 and target 8. So the version bump went to 1.5.0 and I'll target 2.0.0 for Java 11 at some future date.